File: sol/2970/2970.py
class Solution:
    def incremovableSubarrayCount(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1:
            return 1
        isPre = [False] * n  # [0, i)
        isPre[0] = True
        isPre[1] = True
        for i in range(2, n):
            if nums[i - 1] <= nums[i - 2]:
                break
            isPre[i] = True
        isPost = [False] * n  # (j, n - 1]
        isPost[n - 1] = True
        isPost[n - 2] = True
        for i in range(n - 3, -1, -1):
            if nums[i + 1] >= nums[i + 2]:
                break
            isPost[i] = True
        return sum(
            i == 0 or j == n - 1 or nums[i - 1] < nums[j + 1]
            for i in range(n)
            for j in range(i, n)
            if isPre[i] and isPost[j]
        )

File: sol/2970/2970.cpp
struct Solution {
    int incremovableSubarrayCount(vector<int>& nums) {
        int n = nums.size();
        if (n == 1){
            return 1;
        }
        vector<bool> isPre(n, false), isPost(n, false);
        isPre[0] = true; isPre[1] = true;
        for (int i = 2; i < n; ++i){
            if (nums[i - 1] <= nums[i - 2]){
                break;
            }
            isPre[i] = true;
        }
        isPost[n - 1] = true; isPost[n - 2] = true;
        for (int i = n - 3; i >= 0; --i){
            if (nums[i + 1] >= nums[i + 2]){
                break;
            }
            isPost[i] = true;
        }
        int res = 0;
        for (int i = 0; i < n; ++i){
            if (!isPre[i]){
                continue;
            }
            for (int j = i; j < n; ++j){
                if (isPost[j]){
                    res += i == 0 || j == n - 1 || nums[i - 1] < nums[j + 1];
                }
            }
        }
        return res;
    }
};
